### ISSUE TITLE
Attribution for vendored code. 

### ISSUE THREAD

**Issue by Goldziher** [dymmond/esmerald#6]
Hi there,

I see some parts of this code are derived from Starlite - without attribution. Please add the Starlite license where applicable and add a clear attribution. 

Thanks

**Comment by tarsil** [comment]
@Goldziher absolutely. I mentioned Starlite but I wasn't sure where to place it. I will add it! Thank you and keep it up with the great work.

**Comment by Goldziher** [comment]
Kwargs, signature model, parsers ans more. You've copied quite a lot from starlite, so it's only fair to attribute this correctly.

**Comment by tarsil** [comment]
So, I didn't copy perse, I actually refactor a lot. Initially I indeed needed something to help me out with the initial setup. I actually th oughtinitially to build on the top of Starlite but then I understood that you had slightly different plans from the ones I was aiming. Today I refactored a lot of the code that was initially there because it wasn't compatible to what is coming but I always mentioned starlite as inspiration. For instance the way you did the OpenAPI was beautiful and that the whole thing is attributed to Starlite. There are also some things that Starlite took for instance, from Django as initial ideas and the only reason why I know this is because I did a lot of it in the past as I'm sure you did too. 


In the docs you can see Starlite is mentioned and always will because is a great framework with a great community and with great plans. I hear you are even taking away Starlette as it is blocking you from progressing in some points. 

What was core from starlite was indeed the signature and kwargs which I refactored to match Esmerald needs but again, heavily inspired by your great work. 

I'm doing some refactoring in the codebase as I want to remove what I initially did when I was building on the top of Starlite but never removed.

**Comment by PeterChain** [comment]
@Goldziher I remember seeing right at the initial page of the documentation that Esmerald took inspiration from Starlite and implemented on top of Starlette and Pydantic.

**Comment by Goldziher** [comment]
Read the MIT license agreement. It's that simple. Inspiration is one thing, vendoring of code is another. Please make it explicit you are using our code as a foundation. 

A simple comparison shows that huge parts of the codebase are copied from elsewhere, please add the license agreement wherever this applies. 

If you prefer, you can add the license agreement in your license agreement with a note that explains this + clarification in the docs.

**Comment by tarsil** [comment]
I know how the license agreement works and again it states that it should be added (like I did) when copy not significant portions of the code (which it's not there anymore). That also means the copy without changing and direct use. Again, not there anymore. 

I have 0 reasons to vendor code from others and the fact that you are so focused on this without even reading what I said also is showing something else. 

Your license is in the codebase already and you are mentioned in the docs as you were always. That never changed.

Btw,  you think I got the parsers idea from you? No, I went to see your codebase and everything is completely different besides the OpenAPI where I put your license. FastAPI for instance built and ASGI framework, you got inspired and built starlite, I saw a lot of things where recently added to Starletire there that already exist in FastAPI. Hm, did you copy that? Did you got inspired but the work out there ou simply you upgraded like everyone does? I believe it's the latter.

**Comment by Goldziher** [comment]
You can write whatever you want. Git though has memory and diffs are easy to use. When you copy code from elsewhere, it is standard practice to vendor it correctly. This is all I'm asking. You are using our work, and that if others, make it clear and legal. As for bad faith, i had no assumptions, I simply assumed you did not know. This is now changing though, and I am increasingly starting to have a feeling of bad faith on your part. If you did in fact vendored without acknowleding this while understanding this is not how it's done, than yes - we have a problem.

**Comment by tarsil** [comment]
There is nothing to believe. You are correct in your statement about not knowing, I even mentioned that I didn't know where to put it. I was even honest and said that I initially built on the top of your framework :). I'm currently updating some of the codebase to match what is coming and yes, of course I will add the licenses where they need to be and remove what is not being used anymore as it was changed months ago that is why I put your license straight in the OpenAPI codebase when you mentioned. But btw I like your idea. Adding this to my license agreement will actually be better :). How should I phrase it? Any suggestion?

The docs already mentioned you but i cani add some extras to make it even clear.

**Comment by tarsil** [comment]
@Goldziher no worries, I think I understood what you meant :). I will be adding a vendors section and folder, place your license and mention this. Even by inspiration you deserve to be mentioned properly because of it. I can just do it during the week as I'm not currently in my working station if you don't mind?

**Comment by Goldziher** [comment]
> @Goldziher no worries, I think I understood what you meant :). I will be adding a vendors section and folder, place your license and mention this. Even by inspiration you deserve to be mentioned properly because of it. I can just do it during the week as I'm not currently in my working station if you don't mind?

Of course, it's not urgent! Thanks

**Comment by tarsil** [comment]
Keep it up with the great work @Goldziher . You did something truly remarkable there that inspired me to build something good as well :)

**Comment by Goldziher** [comment]
> Keep it up with the great work @Goldziher . You did something truly remarkable there that inspired me to build something good as well :)

Great, im glad this is resolved. I'll star your repo 🙂

**Comment by tarsil** [comment]
Oh I appreciate that. It is. My bad I didn't get it initially here. Next week you should see a vendors folder and your license there + docs. Have a great one @Goldziher
